conf/nginx/nginx.conf.web.https.default.template

```diff
@@ -1,24 +1,32 @@
 # HTTPS Proxy Default Configuration
-#
-server {
-    ${core.ipboth.enabled}listen                  [::]:${web.https.port} default_server ipv6only=off;
-    ${core.ipv4only.enabled}listen                ${web.https.port} default_server;
-    ${core.ipv6only.enabled}listen                [::]:${web.https.port} default_server;
 
-    ssl                     on;
-    ssl_protocols           ${web.ssl.protocols};
-    ssl_prefer_server_ciphers ${web.ssl.preferserverciphers};
-    ssl_session_cache       ${ssl.session.cachesize};
-    ssl_session_timeout     ${ssl.session.timeout};
-    ssl_ciphers             ${web.ssl.ciphers};
-    ssl_ecdh_curve          ${web.ssl.ecdh.curve};
-    ssl_certificate         ${ssl.crt.default};
-    ssl_certificate_key     ${ssl.key.default};
-    ssl_verify_client       ${ssl.clientcertmode.default};
-    ssl_verify_depth        ${ssl.clientcertdepth.default};
-    ${web.ssl.dhparam.enabled}ssl_dhparam             ${web.ssl.dhparam.file};
-    return 444;
-}
+${web.strict.servername}# Strict servername enforcing block
+${web.strict.servername}# Enabled/disabled through the 'zimbraReverseProxyStrictServerName' configuration item
+${web.strict.servername}# The $\{listen.:addresses\} is NOT demarcated with web.strict.servername on purpose.
+${web.strict.servername}server {
+${web.strict.servername}    ${core.ipboth.enabled}listen                  [::]:${web.https.port} default_server ipv6only=off;
+${web.strict.servername}    ${core.ipv4only.enabled}listen                ${web.https.port} default_server;
+${web.strict.servername}    ${core.ipv6only.enabled}listen                [::]:${web.https.port} default_server;
+${web.strict.servername}
+${web.strict.servername}
+${web.strict.servername}    # Listen addresses extracted from `zimbraVirtualIPAddress` on each domain
+${listen.:addresses}
+${web.strict.servername}    # Listen addresses extracted from `zimbraVirtualIPAddress` on each domain
+${web.strict.servername}
+${web.strict.servername}    ssl                     on;
+${web.strict.servername}    ssl_protocols           ${web.ssl.protocols};
```

Comment:
Given the repetition of numerous TLS parameters, I'd like to see us pull the common things out into a separate ```include our/common/tls/settings``` file .  For example:
```
zimbraReverseProxySSLProtocols ( ssl_protocols / proxy_ssl_protocols / ${web.ssl.protocols} )
zimbraReverseProxySSLSessionCacheSize
zimbraReverseProxySSLCiphers
```
etc., shows up in a consistent manner/usage in all of these templates:

> nginx.conf.web.admin.default.template
> nginx.conf.web.admin.template
> nginx.conf.web.https.default.template
> nginx.conf.web.https.template
> nginx.conf.web.sso.default.template
> nginx.conf.web.sso.template

Reply:
Is this not outside the scope of this change?
I’d like to open another ticket to address this kind of change.

Reply:
ZCS-3353 was opened to address the SSL consolidation.